SpatialGDK/Source/SpatialGDK/Private/EngineClasses/SpatialVirtualWorkerTranslator.cpp

```diff
@@ -231,27 +255,33 @@ void SpatialVirtualWorkerTranslator::QueryForWorkerEntities()
 // returned information will be thrown away.
 void SpatialVirtualWorkerTranslator::WorkerEntityQueryDelegate(const Worker_EntityQueryResponseOp& Op)
 {
+	bWorkerEntityQueryInFlight = false;
+
 	check(StaticComponentView.IsValid());
 	if (!StaticComponentView->HasAuthority(SpatialConstants::INITIAL_VIRTUAL_WORKER_TRANSLATOR_ENTITY_ID, SpatialConstants::VIRTUAL_WORKER_TRANSLATION_COMPONENT_ID))
 	{
 		UE_LOG(LogSpatialVirtualWorkerTranslator, Log, TEXT("(%s) Received response to WorkerEntityQuery, but don't have authority over VIRTUAL_WORKER_MANAGER_COMPONENT.  Aborting processing."), *WorkerId);
+		return;
 	}
 	else if (Op.status_code != WORKER_STATUS_CODE_SUCCESS)
 	{
-		UE_LOG(LogSpatialVirtualWorkerTranslator, Log, TEXT("(%s) Could not find Worker Entities via entity query: %s"), *WorkerId, UTF8_TO_TCHAR(Op.message));
-	}
-	else if (Op.result_count == 0)
-	{
-		UE_LOG(LogSpatialVirtualWorkerTranslator, Log, TEXT("(%s) Worker Entity query shows that Worker Entities do not yet exist in the world."), *WorkerId);
+		UE_LOG(LogSpatialVirtualWorkerTranslator, Warning, TEXT("(%s) Could not find Worker Entities via entity query: %s, retrying."), *WorkerId, UTF8_TO_TCHAR(Op.message));
 	}
 	else
 	{
 		UE_LOG(LogSpatialVirtualWorkerTranslator, Log, TEXT("(%s) Processing Worker Entity query response"), *WorkerId);
 		ConstructVirtualWorkerMappingFromQueryResponse(Op);
-		SendVirtualWorkerMappingUpdate();
 	}
 
-	bWorkerEntityQueryInFlight = false;
+	// If the translation mapping is complete, publish it. Otherwise retry the worker entity query.
```

Comment:
there is no retrying here?

Reply:
Good catch. I was seeing extra updates in my testing and one of the things I tried was removing the retry, then I forgot to add it back. Got it now.